_component_design/date-picker.md

```diff
@@ -0,0 +1,341 @@
+
+In this design the field width is extended to match the width of the picker, and it contains both the start and end date. The picker itself allows a range to be selected, filling in the field as the user makes choices.
+
+**Selection logic is as follows:**
+ * The user’s first click on a date sets the start date value
+ * The user’s second click sets the end-date value, unless it is before the start date; then it just sets a new start date
+ * Dates in-between are highlighted as shown
+
+**Selections made after the start or end have been chosen follow this logic:**
+ * A selection AFTER the end date just extends the range, setting the end date to the new value
+ * A selection BEFORE the start date is like starting over, clearing the entire selection and filling in a new start date.
+{% endcolumn %}
+
+{% column right:"hxCol-8 hxCol-xs-12 hxCol-sm-12 hxCol-md-8 hxCol-lg-8" %}
+
+<div class="hxRow">
+{% column right:"hxCol-12" %}
+{% figure [class:"image bg-light border"] %}
+![]({{site.url}}/assets/images/components/inputs-and-controls/date-picker/date-picker-date-ranges-using-a-single-field-1.svg)
+{% endfigure %}
+{% endcolumn %}
+</div>
+<div class="hxRow">
+{% column right:"hxCol-6" %}
+{% figure [caption:"Start date selected, with hover showing range"] [class:"image bg-light border"] %}
+![]({{site.url}}/assets/images/components/inputs-and-controls/date-picker/date-picker-date-ranges-using-a-single-field-2.svg)
+{% endfigure %}
+{% endcolumn %}
+{% column right:"hxCol-6" %}
+{% figure [caption:"End date selected, range is complete"] [class:"image bg-light border"] %}
+![]({{site.url}}/assets/images/components/inputs-and-controls/date-picker/date-picker-date-ranges-using-a-single-field-3.svg)
+{% endfigure %}
+{% endcolumn %}
+</div>
+
+{% endcolumn %}
+</div>
+
+<div class="hxRow">
+{% column left:"hxCol-4 hxCol-xs-12 hxCol-sm-12 hxCol-md-4 hxCol-lg-4" %}
+
+#### Ranges spanning months or years
+If the user selects a start date in one month or year, and an end date in another month or year, all dates in-between highlight as expected.
+
+{% endcolumn %}
+
+{% column right:"hxCol-4 hxCol-xs-6 hxCol-sm-6 hxCol-md-4 hxCol-lg-4" %}
+{% figure [caption:"Start date selected in October"] [class:"image bg-light border"] %}
+![]({{site.url}}/assets/images/components/inputs-and-controls/date-picker/date-picker-ranges-spanning-months-or-years-1.svg)
+{% endfigure %}
+{% endcolumn %}
+
+{% column right:"hxCol-4 hxCol-xs-6 hxCol-sm-6 hxCol-md-4 hxCol-lg-4" %}
+{% figure [caption:"End date selected in December"] [class:"image bg-light border"] %}
+![]({{site.url}}/assets/images/components/inputs-and-controls/date-picker/date-picker-ranges-spanning-months-or-years-2.svg)
+{% endfigure %}
+{% endcolumn %}
+</div>
+
+## Month picker
+<div class="hxRow">
+{% column left:"hxCol-8 hxCol-xs-12 hxCol-sm-12 hxCol-md-8 hxCol-lg-8" %}
```

Comment:
I'd change this column to full width since the pictures are on their own line.
```
hxCol-12 hxCol-xs-12 hxCol-sm-12 hxCol-md-12 hxCol-lg-12
```

Also, before I forget, you can skip sizes if it makes things smaller and easier to deal with. I forget which way it goes in terms of whether you remove the smaller one of two duplicates (e.g., whether you'd remove `xs` or `sm` here to get the same thing)... need to document that.

Reply:
@nimbinatus this was done to prevent the text from spanning the whole width of the container.  On a previous episode of Helix, we talked about limiting the width of overhead text blocks like this so they didn't look all strung out.